doppelganger/doppelganger.go

```diff
@@ -292,20 +306,30 @@ func (h *handler) processLivenessData(epoch phase0.Epoch, livenessData []*eth2ap
 				fields.ValidatorIndex(response.Index),
 				zap.Uint64("remaining_epochs", uint64(state.remainingEpochs)))
 		} else {
+			validatorsStateCounter.Add(ctx, -1, metric.WithAttributes(unsafeAttribute(true)))
+			validatorsStateCounter.Add(ctx, 1, metric.WithAttributes(unsafeAttribute(false)))
```

Comment:
i think there could be a race condition case where the validator was unsafe to begin with, but by the time the liveness request finished we've already observed quorum for it and incremented the metric, only to then increment it again here

wdyt?

Reply:
if that's true, i wouldn't rush to solve this

i think we can come up with something which fundamentally prevents this and similar issues

Reply:
Maybe something simpler like just updating all the states in every slot..

Reply:
Good catch, I switched to `Gauge` and now update all states once per slot to avoid this race.